assets/src/scripts/components/Search.js

```diff
@@ -0,0 +1,115 @@
+import 'whatwg-fetch';
+import debounce from 'lodash/debounce';
+import template from 'lodash/template';
+
+var init = function( searchBar ) {
+
+	var searchField      = searchBar.querySelector( '#site-search' ),
+	    resultsContainer = searchBar.querySelector( '#site-search-results' );
```

Comment:
The reason I used querySelector, is that it allows you to only select children of searchBar, but getElementById needs to be document wide.

I think i'm going to switch to a class. There's no reason you couldn't have multiple search fields on a page, and the JS implementation pretty much allows for that already, so lets not tie it to an ID.